### PR TITLE
Updates on components.rst

### PR DESCRIPTION
**Pull request by tabergma**
**Proposed changes**:
- part of https://github.com/RasaHQ/rasa/issues/5308
- add a collapsable container to hide "advanced" parameters by default
- update description of model parameters of DIETClassifier, etc. 
- keep documentation more consistent

**Status (please check what you already did)**:
- [ ] added some tests for the functionality
- [x] updated the documentation
- [ ] updated the changelog (please check [changelog](https://github.com/RasaHQ/rasa/tree/master/changelog) for instructions)
- [ ] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa#code-style) for instructions)

### CODE REVIEW

File: docs/nlu/components.rst [hunk 1]
```diff
@@ -1458,204 +1500,161 @@ DIETClassifier
...
-            # If 'True' random tokens of the input message will be masked and the model
-            # should predict those tokens.
-            "use_masked_language_model": False
-            # 'BILOU_flag' determines whether to use BILOU tagging or not.
-            # If set to 'True' labelling is more rigorous, however more
-            # examples per entity are required.
-            # Rule of thumb: you should have more than 100 examples per entity.
-            "BILOU_flag": True
+    You can define a number of hyperparameters to adapt the model.
+    If you want to adapt your model, start by modifying the following parameters:
```
Comment (tabergma) [inline docs/nlu/components.rst]: @Ghostvv What parameters do you think users should modify first when adapting the model?

Comment (Ghostvv) [reply docs/nlu/components.rst]: epochs

Comment (Ghostvv) [reply docs/nlu/components.rst]: number of transformer layers,
transformer_size,
hidden_layers_sizes,
embedding_dimension

Comment (Ghostvv) [reply docs/nlu/components.rst]: I'm not sure

Comment (tabergma) [reply docs/nlu/components.rst]: yeah, also not sure, will add 
transformer_size,
hidden_layers_sizes,
embedding_dimension
thanks

Comment (Ghostvv) [reply docs/nlu/components.rst]: let's also put `number of transformer layers` in there, that's the parameter that people probably are most familiar with

Comment (tabergma) [reply docs/nlu/components.rst]: is already in there, just updated

**Comment by tabergma** [comment]
I would recommend to checkout the branch and build the docs (e.g. `make livedocs`) otherwise it is hard to see how the toggle actual works.

File: docs/nlu/components.rst [hunk 2]
```diff
@@ -765,160 +813,127 @@ EmbeddingIntentClassifier
...
+          This parameter allows you to define the number of feed forward neural network layers and their output
+          dimensions for user messages and intents (default: ``text: [], label: []``).
+          Every entry in the list corresponds to a feed forward neural network layer.
+          For example, if you set ``text: [256, 128]``, we will add two feed forward neural network layers in front of
+          the transformer. The vectors of the input tokens (coming from the user message) will be passed on to those
+          layers. The first layer will have an output dimension of 256 and the second layer will have an output
+          dimension of 128. If an empty list is used (default behaviour), no feed forward neural network layer will be
+          added.
+          Make sure to use only positive integer values. Usually, numbers of power of two are used. Also, the integer
+          values in a list should decrease, values in the list should be smaller or equal the value before.
```
Comment (Ghostvv) [inline docs/nlu/components.rst]:
```suggestion
          Also, it is usual practice to have decreasing values in the list: next value is smaller or equal to the value before.
```

File: docs/nlu/components.rst [hunk 3]
```diff
@@ -992,114 +1007,168 @@ ResponseSelector
...
+        - ``epochs``:
+          This parameter sets the number of times the algorithm will see the training data (default: ``300``).
+          One ``epoch`` is equals to one forward pass and one backward pass of all the training examples.
+          Sometimes the model needs more epochs to properly learn.
+          Sometimes more epochs don't influence the performance.
+          The lower the number of epochs the faster the model is trained.
+        - ``hidden_layers_sizes``:
+          This parameter allows you to define the number of feed forward neural network layers and their output
+          dimensions for user messages and intents (default: ``text: [], label: []``).
+          Every entry in the list corresponds to a feed forward neural network layer.
```
Comment (Ghostvv) [inline docs/nlu/components.rst]:
```suggestion
          Every entry in the list corresponds to a feed forward layer.
```

File: docs/nlu/components.rst [hunk 4]
```diff
@@ -992,114 +1007,168 @@ ResponseSelector
...
+    If you want to adapt your model, start by modifying the following parameters:
+
+        - ``epochs``:
+          This parameter sets the number of times the algorithm will see the training data (default: ``300``).
+          One ``epoch`` is equals to one forward pass and one backward pass of all the training examples.
+          Sometimes the model needs more epochs to properly learn.
+          Sometimes more epochs don't influence the performance.
+          The lower the number of epochs the faster the model is trained.
+        - ``hidden_layers_sizes``:
+          This parameter allows you to define the number of feed forward neural network layers and their output
```
Comment (Ghostvv) [inline docs/nlu/components.rst]:
```suggestion
          This parameter allows you to define the number of feed forward layers and their output
```

File: docs/nlu/components.rst [hunk 5]
```diff
@@ -992,114 +1007,168 @@ ResponseSelector
...
+          dimensions for user messages and intents (default: ``text: [], label: []``).
+          Every entry in the list corresponds to a feed forward neural network layer.
+          For example, if you set ``text: [256, 128]``, we will add two feed forward neural network layers in front of
```
Comment (Ghostvv) [inline docs/nlu/components.rst]:
```suggestion
          For example, if you set ``text: [256, 128]``, we will add two feed forward layers in front of
```

File: docs/nlu/components.rst [hunk 6]
```diff
@@ -992,114 +1007,168 @@ ResponseSelector
...
+          the transformer. The vectors of the input tokens (coming from the user message) will be passed on to those
+          layers. The first layer will have an output dimension of 256 and the second layer will have an output
+          dimension of 128. If an empty list is used (default behaviour), no feed forward neural network layer will be
+          added.
+          Make sure to use only positive integer values. Usually, numbers of power of two are used. Also, the integer
+          values in a list should decrease, values in the list should be smaller or equal the value before.
```
Comment (Ghostvv) [inline docs/nlu/components.rst]: same as for embedding intent classifier

File: docs/nlu/components.rst [hunk 7]
```diff
@@ -1458,204 +1527,168 @@ DIETClassifier
...
+    If you want to adapt your model, start by modifying the following parameters:
+
+        - ``epochs``:
+          This parameter sets the number of times the algorithm will see the training data (default: ``300``).
+          One ``epoch`` is equals to one forward pass and one backward pass of all the training examples.
+          Sometimes the model needs more epochs to properly learn.
+          Sometimes more epochs don't influence the performance.
+          The lower the number of epochs the faster the model is trained.
+        - ``hidden_layers_sizes``:
+          This parameter allows you to define the number of feed forward neural network layers and their output
```
Comment (Ghostvv) [inline docs/nlu/components.rst]: `feed forward neural network layers` should be changed to `feed forward layers`.
`network` here would be confusing

File: docs/nlu/components.rst [hunk 8]
```diff
@@ -1458,204 +1527,168 @@ DIETClassifier
...
+          dimensions for user messages and intents (default: ``text: [], label: []``).
+          Every entry in the list corresponds to a feed forward neural network layer.
+          For example, if you set ``text: [256, 128]``, we will add two feed forward neural network layers in front of
+          the transformer. The vectors of the input tokens (coming from the user message) will be passed on to those
+          layers. The first layer will have an output dimension of 256 and the second layer will have an output
+          dimension of 128. If an empty list is used (default behaviour), no feed forward neural network layer will be
+          added.
+          Make sure to use only positive integer values. Usually, numbers of power of two are used. Also, the integer
+          values in a list should decrease, values in the list should be smaller or equal the value before.
```
Comment (Ghostvv) [inline docs/nlu/components.rst]: same as in embedding intent classifier

**Comment by Ghostvv** [comment]
@tabergma I'd also add `weight_sparsity` for people to change

File: docs/nlu/components.rst [hunk 9]
```diff
@@ -107,46 +102,52 @@ HFTransformersNLP
      .. note:: To use ``HFTransformersNLP`` component, install Rasa Open Source with ``pip install rasa[transformers]``.
 
 :Configuration:
+    You should specify what language model to load via the parameter ``model_name``. See the below table for the
+    available language models.
+    Additionally, you can also specify the the architecture variation of the chosen language model by specifying the
+    parameter ``model_weights``.
+    The full list of supported architectures can be found
+    `here <https://huggingface.co/transformers/pretrained_models.html>`__.
```
[the remark this answers is not in the document]
Comment (dakshvar22) [reply docs/nlu/components.rst]: @Ghostvv We support all models that we show in our list of supported language model tables(https://github.com/RasaHQ/rasa/blob/1fa841c4a1e3a4ad178f7cf05c619184eded42e1/docs/nlu/components.rst#hftransformersnlp). The link is for pointing the developer to the table which lists all model variants of each language model.

**Comment by tabergma** [comment]
@erohmensing @ArjaanBuijk Any feedback?

File: docs/nlu/components.rst [hunk 10]
```diff
@@ -39,9 +32,9 @@ MitieNLP
 :Outputs: Nothing
 :Requires: Nothing
 :Description:
-    Initializes mitie structures. Every mitie component relies on this,
+    Initializes MITIE structures. Every MITIE component relies on this,
     hence this should be put at the beginning
-    of every pipeline that uses any mitie components.
+    of every pipeline that uses any MITIE components.
```
Comment (ArjaanBuijk) [inline docs/nlu/components.rst]: What happens if you forget to specify this? Do you get an error?

File: docs/nlu/components.rst [hunk 11]
```diff
@@ -107,50 +102,56 @@ HFTransformersNLP
      .. note:: To use ``HFTransformersNLP`` component, install Rasa Open Source with ``pip install rasa[transformers]``.
 
 :Configuration:
+    You should specify what language model to load via the parameter ``model_name``. See the below table for the
+    available language models.
+    Additionally, you can also specify the the architecture variation of the chosen language model by specifying the
```
Comment (ArjaanBuijk) [inline docs/nlu/components.rst]:
```suggestion
    Additionally, you can also specify the architecture variation of the chosen language model by specifying the
```

File: docs/nlu/components.rst (hunk 10 above)
Comment (tabergma) [reply docs/nlu/components.rst]: Yes. If a required component is not in your pipeline, you will get an error pointing you to the missing component.

File: docs/nlu/components.rst [hunk 12]
```diff
@@ -769,169 +818,144 @@ EmbeddingIntentClassifier
...
+
+         +---------------------------------+------------------+--------------------------------------------------------------+
+         | Parameter                       | Default Value    | Description                                                  |
+         +=================================+==================+==============================================================+
+         | hidden_layers_sizes             | text: [256, 128] | Hidden layer sizes for layers before the embedding layers    |
+         |                                 | label: []        | for user messages and labels. The number of hidden layers is |
+         |                                 |                  | equal to the length of the corresponding.                    |
+         +---------------------------------+------------------+--------------------------------------------------------------+
+         | share_hidden_layers             | False            | Whether to share the hidden layer weights between user       |
+         |                                 |                  | messages and labels.                                          |
```
Comment (erohmensing) [inline docs/nlu/components.rst]:
```suggestion
         |                                 |                  | messages and labels.                                         |
```

File: docs/nlu/components.rst [hunk 13]
```diff
@@ -1005,123 +1029,184 @@ ResponseSelector
...
-    In addition, the component can also be configured to train a response selector for a particular retrieval intent.
-
-        - ``retrieval_intent`` sets the name of the intent for which this response selector model is trained.
-          Default is ``None``, i.e. the model is trained for all retrieval intents.
-
-    Default values:
-
-    .. code-block:: yaml
+    The algorithm includes almost all the hyperparameters that :ref:`diet-classifier` uses.
+    If you want to adapt your model, start by modifying the following parameters:
```
Comment (erohmensing) [inline docs/nlu/components.rst]: The descriptions of the ones to modify are super helpful!

File: docs/nlu/components.rst [hunk 14]
```diff
@@ -526,55 +536,89 @@ CountVectorsFeaturizer
...
+          "use_shared_vocab": False
+
+    .. container:: toggle
+
+        .. container:: header
+
+            The above configuration parameters are the ones you most likely gonna change.
+            However, additional parameters exists that can be adapted.
+
+        .. code-block:: none
```
[the remark this answers is not in the document]
Comment (erohmensing) [reply docs/nlu/components.rst]: Maybe even something like 

However, additional parameters exists that can be adapted. *(click to expand)*

File: docs/nlu/components.rst [hunk 15]
```diff
@@ -1326,15 +1408,8 @@ CRFEntityExtractor
...
-            # rule of thumb: use only if more than 100 egs. per entity
             "BILOU_flag": True
-            # crf_features is [before, token, after] array with before, token,
-            # after holding keys about which features to use for each token,
-            # for example, 'title' in array before will have the feature
-            # "is the preceding token in title case?"
-            # POS features require SpacyTokenizer
-            # pattern feature require RegexFeaturizer
+            # features to extract in the sliding window
             "features": [
```
Comment (erohmensing) [inline docs/nlu/components.rst]: I think we need to un-tab "bilou_flag" and "features" to match where "name" is

File: docs/nlu/components.rst [hunk 16]
```diff
@@ -1359,6 +1434,12 @@ CRFEntityExtractor
             # weight of the L2 regularization
             "L2_c": 0.1
 
+    .. note::
+        If POS features are used (pos or pos2), you need to have ``SpacyTokenizer`` in your pipeline.
+
+    .. note::
+        If "pattern" features are used, you need to have ``RegexFeaturizer`` in your pipeline.
+
```
Comment (erohmensing) [inline docs/nlu/components.rst]: code-format `"pos"` `"pos2"`, `"pattern"`

File: docs/nlu/components.rst [hunk 17]
```diff
@@ -1480,213 +1565,179 @@ DIETClassifier
...
-            # You can view the training metrics after training in tensorboard via
-            # ``tensorboard --logdir <path-to-given-directory>``
-            "tensorboard_log_directory": None
-            # Define when training metrics for tensorboard should be logged.
-            # Either after every epoch or for every training step.
-            # Valid values: 'epoch' and 'minibatch'
-            "tensorboard_log_level": "epoch"
+    If you want to use the ``DIETClassifier`` just for intent classification, set ``entity_recognition`` to ``False``.
+    If you want to do only entity recognition, set ``intent_classification`` to ``False``.
+    By default ``DIETClassifier`` is doing both, i.e. ``entity_recognition`` and ``intent_classification`` is set to
```
Comment (erohmensing) [inline docs/nlu/components.rst]:
```suggestion
    By default ``DIETClassifier`` does both, i.e. ``entity_recognition`` and ``intent_classification`` are set to
```

File: docs/nlu/components.rst [hunk 18]
```diff
@@ -526,55 +536,89 @@ CountVectorsFeaturizer
...
+          "OOV_token": "_oov_"
+          # Whether to use a shared vocab
+          "use_shared_vocab": False
+
+    .. container:: toggle
+
+        .. container:: header
+
+            The above configuration parameters are the ones you most likely gonna change.
+            However, additional parameters exists that can be adapted.
```
Comment (erohmensing) [inline docs/nlu/components.rst]:
```suggestion
            The above configuration parameters are the ones you should configure to fit your model to your data.
            However, additional parameters exist that can be adapted.
```
too much? let's change this everywhere when we decide what to say

File: docs/_static/css/custom.css [hunk 19]
```diff
@@ -6,3 +6,18 @@ dl.glossary dt {
...
+}
+
+.toggle .header:after {
+    content: "▶ show";
+    cursor: pointer;
+}
+
+.toggle .header.open:after {
+    content: "▼ hide";
+    cursor: pointer;
```
Comment (erohmensing) [inline docs/_static/css/custom.css]: I guess these are supposed to show? they're not rendering for me, but i've had issues with my CSS rendering in `make livedocs` before. does it reneder on livedoc for you?

File: docs/nlu/components.rst [hunk 20]
```diff
@@ -487,12 +497,14 @@ CountVectorsFeaturizer
...
     By default ``analyzer`` is set to ``word`` so word token counts are used as features.
     If you want to use character n-grams, set ``analyzer`` to ``char`` or ``char_wb``.
+    The lower and upper boundaries of the n-grams can be configured via the parameters ``min_ngram`` and ``max_ngram``.
+    By default both of them are set to ``1``.
 
     .. note::
-        Option ‘char_wb’ creates character n-grams only from text inside word boundaries;
+        Option ``char_wb`` creates character n-grams only from text inside word boundaries;
         n-grams at the edges of words are padded with space.
         This option can be used to create `Subword Semantic Hashing <https://arxiv.org/abs/1810.07150>`_.
```
[the remark this answers is not in the document]
Comment (tabergma) [reply docs/nlu/components.rst]: yes

File: docs/nlu/components.rst (hunk 18 above)
Comment (erohmensing) [reply docs/nlu/components.rst]: This is my last open comment -- regardless of how we want to phrase this, in the 3 places we say this we should change
`gonna` -> `going to`
`exists` -> `exist`

I thought it was weird to say that someone is "most likely going to change" something, cause we have no idea what the'll do, but we can also leave it like that as long as the grammar is fixed 🤷‍♀️

**Pull request merged**